python/cuml/cuml/tests/test_spectral_embedding.py

```diff
@@ -0,0 +1,260 @@
+    X, y = mnist.data, mnist.target.astype(np.int32)
+
+    # Normalize and sample
+    X = X / 255.0
+    X, _, y, _ = train_test_split(
+        X, y, train_size=n_samples, stratify=y, random_state=42
+    )
+
+    # sklearn embedding
+    sk_spectral = skSpectralEmbedding(
+        n_components=N_COMPONENTS,
+        n_neighbors=N_NEIGHBORS,
+        affinity="nearest_neighbors",
+        random_state=42,
+        n_jobs=-1,
+    )
+    X_sklearn = sk_spectral.fit_transform(X)
+
+    # cuML embedding
+    X_gpu = cp.asarray(X)
+    cuml_spectral = SpectralEmbedding(
+        n_components=N_COMPONENTS, n_neighbors=N_NEIGHBORS, random_state=42
+    )
+    X_cuml_gpu = cuml_spectral.fit_transform(X_gpu)
+    X_cuml = cp.asnumpy(X_cuml_gpu)
+
+    # Calculate trustworthiness scores
+    trust_sklearn = trustworthiness(X, X_sklearn, n_neighbors=N_NEIGHBORS)
+    trust_cuml = trustworthiness(X, X_cuml, n_neighbors=N_NEIGHBORS)
+
+    print(f"\nMNIST (n={n_samples}):")
+    print(f"  sklearn trustworthiness: {trust_sklearn:.4f}")
+    print(f"  cuML trustworthiness: {trust_cuml:.4f}")
+    print(f"  Difference: {abs(trust_sklearn - trust_cuml):.4f}")
+
+    # Both should have reasonable trustworthiness (> 0.8 for MNIST)
+    assert (
+        trust_sklearn > 0.8
+    ), f"sklearn trustworthiness {trust_sklearn:.4f} is too low"
+    assert (
+        trust_cuml > 0.8
+    ), f"cuML trustworthiness {trust_cuml:.4f} is too low"
+
+
+@pytest.mark.parametrize("n_samples", [5000])
+def test_spectral_embedding_trustworthiness_fashion_mnist(n_samples):
+    """Test trustworthiness comparison between sklearn and cuML on Fashion-MNIST dataset."""
+    # Load Fashion-MNIST data
+    X = load_fashion_mnist_data(max_samples=n_samples)
+
+    # sklearn embedding
+    sk_spectral = skSpectralEmbedding(
+        n_components=N_COMPONENTS,
+        n_neighbors=N_NEIGHBORS,
+        affinity="nearest_neighbors",
+        random_state=42,
+        n_jobs=-1,
+    )
+    X_sklearn = sk_spectral.fit_transform(X)
+
+    # cuML embedding
+    X_gpu = cp.asarray(X)
+    cuml_spectral = SpectralEmbedding(
+        n_components=N_COMPONENTS, n_neighbors=N_NEIGHBORS, random_state=42
+    )
+    X_cuml_gpu = cuml_spectral.fit_transform(X_gpu)
+    X_cuml = cp.asnumpy(X_cuml_gpu)
+
+    # Calculate trustworthiness scores
+    trust_sklearn = trustworthiness(X, X_sklearn, n_neighbors=N_NEIGHBORS)
+    trust_cuml = trustworthiness(X, X_cuml, n_neighbors=N_NEIGHBORS)
+
+    print(f"\nFashion-MNIST (n={n_samples}):")
+    print(f"  sklearn trustworthiness: {trust_sklearn:.4f}")
+    print(f"  cuML trustworthiness: {trust_cuml:.4f}")
+    print(f"  Difference: {abs(trust_sklearn - trust_cuml):.4f}")
+
+    # Both should have reasonable trustworthiness (> 0.8 for Fashion-MNIST)
+    assert (
+        trust_sklearn > 0.8
+    ), f"sklearn trustworthiness {trust_sklearn:.4f} is too low"
+    assert (
+        trust_cuml > 0.8
+    ), f"cuML trustworthiness {trust_cuml:.4f} is too low"
```

Comment:
pytest does nice logging for assert failures by default, no need to add extra formatted strings, the variable names and values will be present in the logs by default anyway.

```suggestion
    assert trust_sklearn > 0.8
    assert trust_cuml > 0.8
```

Please apply this tip everywhere else in this file too.